charts/capsule-proxy/templates/certmanager.yaml

```diff
@@ -0,0 +1,50 @@
+{{- if and .Values.certManager.generateCertificates .Values.options.enableSSL -}}
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: {{ include "capsule-proxy.fullname" . }}-selfsigned-issuer
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ include "capsule-proxy.fullname" . }}-selfsigned-ca
+spec:
+  isCA: true
+  commonName: {{ include "capsule-proxy.fullname" . }}-selfsigned-ca
+  secretName: {{ include "capsule-proxy.fullname" . }}-root-secret
+  privateKey:
+    algorithm: ECDSA
+    size: 256
+  issuerRef:
+    name: {{ include "capsule-proxy.fullname" . }}-selfsigned-issuer
+    kind: Issuer
+    group: cert-manager.io
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: {{ include "capsule-proxy.fullname" . }}-ca-issuer
+spec:
+  ca:
+    secretName: {{ include "capsule-proxy.fullname" . }}-root-secret
```

Comment:
There are use cases where we may want to use an existing CA - say for example, the Kubernetes CA. Could we add some flag to enable/disable the self-signed CA certs and allow the user to provide the secret name used in the CA Issuer via the chart values i.e. the one at `spec.ca.secretName` above?

E.g.

```
certManager:
  generateCertificates: true
  existingCA:
    enabled: true
    secretName: my-custom-ca-cert
```

Reply:
Thinking about it, `externalCA` may be a better name than `existingCA`. Just to add, the `certManager.externalCA.secretName` (or whatever we call it) would need to point to a secret containing the CA cert and private key of the external CA in the correct cert-manager format as per https://cert-manager.io/docs/configuration/ca/#deployment

Reply:
A helper could be implemented, something as the following:

```
{{- define "capsule-proxy.caSecretName" }}
{{- if .Values.ca.secretName }}
{{- printf "%s" .Values.ca.secretName }}
{{- else }}
{{- printf "%s-root-secret" (include "capsule-proxy.fullname" .) }}
{{- end }}
```